src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/ExplainTest.java

```diff
@@ -0,0 +1,128 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.unittest;
+
+import com.alibaba.druid.sql.parser.ParserException;
+import com.amazon.opendistroforelasticsearch.sql.esintgtest.TestUtils;
+import com.amazon.opendistroforelasticsearch.sql.esintgtest.TestsConstants;
+import com.amazon.opendistroforelasticsearch.sql.exception.SqlParseException;
+import com.amazon.opendistroforelasticsearch.sql.utils.JsonPrettyFormatter;
+import com.amazon.opendistroforelasticsearch.sql.query.ESActionFactory;
+import com.amazon.opendistroforelasticsearch.sql.query.QueryAction;
+import com.amazon.opendistroforelasticsearch.sql.query.SqlElasticRequestBuilder;
+import com.amazon.opendistroforelasticsearch.sql.request.SqlRequest;
+import com.amazon.opendistroforelasticsearch.sql.util.CheckScriptContents;
+import com.google.common.io.Files;
+import org.elasticsearch.client.Client;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.sql.SQLFeatureNotSupportedException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+public class ExplainTest {
+
+    @Test
+    public void assertExplainRequestShouldBeJson() throws IOException {
+
+        String expectedOutputFilePath = TestUtils.getResourceFilePath(
+                "src/test/resources/expectedOutput/explain_format_pretty.json");
+        String expectedOutput = Files.toString(new File(expectedOutputFilePath), StandardCharsets.UTF_8)
+                .replaceAll("\r", "");
+
+        String explainResult = explain(String.format("{\"query\":\"" +
```

Comment:
Couple comments here:

1. This is not integration test, so no need to use index names here from test constants here. You could just put any index name

2. Without need of using test constants you don't need string formatting

3. In case if you **need** to use string formatting, you need to use StringUtils.format that will use proper locale

Reply:
Thank you for pointing out. I'll modify the IT by today.

Reply:
I've added one more IT and rewrote the UT. Instead of testing the `_explain` request, I went more specific into testing the formatter class, including the functionality and the exception.